addons/calendar/wizard/calendar_popover_delete_wizard.py

```diff
@@ -1,24 +1,93 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class CalendarPopoverDeleteWizard(models.TransientModel):
+    _inherit = ['mail.composer.mixin']
     _description = 'Calendar Popover Delete Wizard'
 
 
     record = fields.Many2one('calendar.event', 'Calendar Event')
     delete = fields.Selection([('one', 'Delete this event'), ('next', 'Delete this and following events'), ('all', 'Delete all the events')], default='one')
+    recipient_ids = fields.Many2many(
+        'res.partner',
+        string="Recipients",
+        compute='_compute_recipient_ids',
+        readonly=False,
+    )
 
     def close(self):
-        if not self.record or not self.delete:
-            pass
-        elif self.delete == 'one':
-            self.record.unlink()
+        if self.record.attendees_count == 1 and self.record.user_id.partner_id == self.record.partner_ids:
```

Comment:
You should go in the opposite direction for not re-writing the entire function:
1. Check if the attendees count is != 1 OR if the organizer partner_id is different than partner_ids, if yes: 
``` return self.record.action_unlink_event(self.record.partner_id.id, self.delete)```
2. Let the following code (after the condition above) function as it is, i.e. the previous code intact.

This will remove all the new diff from the function.